Testing

ChelseaLura opened the pull request:
Not perfectly polished yet; however has logically split-up back-end vs front-testing, as well as unit vs integration testing.

ChelseaLura committed:
Subtle edit to ReadMe to test for proper branching from master

ChelseaLura committed:
Added full package list for Karma (Will add protractor once Karma is fully set-up)

ChelseaLura committed:
Added Karma; spruced up Gruntfile; added simple front-end test for set-up testing.

aisapatino commented on Gruntfile.js:
```diff
@@ -1,19 +1,22 @@
-
+/*jslint node: true */
```
we shouldn't need this since we have a .jshintrc that sets node: true :)

aisapatino commented on app/application.js:
```diff
@@ -1,4 +1,5 @@
 var sleep = require('./modules/sleep/sleep-module.js');
+require('angular');
```
did you see my note in gitter about this? angular should already be bundled & loaded, so this + debowerify might be loading it twice.

aisapatino committed:
fix deps for test-back-all grunt task

aisapatino commented on Gruntfile.js:
```diff
@@ -30,74 +33,85 @@ module.exports = function(grunt) {
 				},
 				files: {
-					'dist/client.js': 'app/application.js'
+					'dist/client.karma.js': ['app/application.js','app/tests/unit/**/*.js']
```
are we moving all front-end tests to a single folder rather than a folder per module?

ChelseaLura committed:
Fixed bugs from pull-request comments

aisapatino committed:
bundle angular files into client.js

loads minified files listed in gruntfile.js into top of client.js
also works with client.karma.js

ChelseaLura committed:
Fixed front-end unit tests. Current build runs all backend tests and front-end unit testing with karma
